packages/foundry/contracts/WitchBase.sol

```diff
@@ -0,0 +1,680 @@
+
+    /*
+
+       x x x
+     x      x    Hi Fren!
+    x  .  .  x   I want to buy this vault under auction!  I'll pay
+    x        x   you in the same `base` currency of the debt, or in fyToken, but
+    x        x   I want no less than `uint min` of the collateral, ok?
+    x   ===  x
+    x       x
+      xxxxx
+        x                             __  Ok Fren!
+        x     ┌────────────┐  _(\    |@@|
+        xxxxxx│ BASE BUCKS │ (__/\__ \--/ __
+        x     │     OR     │    \___|----|  |   __
+        x     │   FYTOKEN  │        \ }{ /\ )_ / _\
+       x x    └────────────┘        /\__/\ \__O (__
+                                   (--/\--)    \__/
+                            │      _)(  )(_
+                            │     `---''---`
+                            ▼
+      _______
+     /  12   \  First lets check how much time `t` is left on the auction
+    |    |    | because that helps us determine the price we will accept
+    |9   |   3| for the debt! Yay!
+    |     \   |                       p + (1 - p) * t
+    |         |
+     \___6___/          (p is the auction starting price!)
+
+                            │
+                            │
+                            ▼                  (\
+                                                \ \
+    Then the Cauldron updates our internal    __    \/ ___,.-------..__        __
+    accounting by slurping up the debt      //\\ _,-'\\               `'--._ //\\
+    and the collateral from the vault!      \\ ;'      \\                   `: //
+                                             `(          \\                   )'
+    The Join then dishes out the collateral    :.          \\,----,         ,;
+    to you, dear user. And the debt is          `.`--.___   (    /  ___.--','
+    settled with the base join or debt fyToken.   `.     ``-----'-''     ,'
+                                                    -.               ,-
+                                                       `-._______.-'
+
+
+    */
+
+    /// @dev quotes how much ink a liquidator is expected to get if it repays an `artIn` amount. Works for both Auctioned and ToBeAuctioned vaults
+    /// @param vaultId The vault to get a quote for
+    /// @param to Address that would get the collateral bought
+    /// @param maxArtIn How much of the vault debt will be paid. GT than available art means all
+    /// @return liquidatorCut How much collateral the liquidator is expected to get
+    /// @return auctioneerCut How much collateral the auctioneer is expected to get. 0 if liquidator == auctioneer
+    /// @return artIn How much debt the liquidator is expected to pay
+    function calcPayout(
+        bytes12 vaultId,
+        address to,
+        uint256 maxArtIn
+    )
+        external
+        view
+        virtual
+        returns (
+            uint256 liquidatorCut,
+            uint256 auctioneerCut,
+            uint256 artIn
+        )
+    {}
```

Comment:
I don't believe you need this, each witch will be forced to have it as it should be on each one of the interfaces